pull.go

```diff
@@ -215,13 +222,10 @@ func pullImage(ctx context.Context, store storage.Store, imageName string, optio
 	spec := imageName
 	srcRef, err := alltransports.ParseImageName(spec)
 	if err != nil {
```

Comment:
The two callers of `pullImage` now explicitly know whether the input should be readable by `ParseImageName` or whether it needs the transport added (and using the other path can misinterpret the input); it would be nice to make this unambiguous.

`pullAndFindImage` AFAICS currently always submits input input with the “concatenate `options.transport`+`imageName`” format, while the `AllTags` path in `Pull` always submits input with the “use `ParseImageName(imageName)` semantics — so, the concatenation code could be moved to `pullAndFindImage` (or maybe even higher to `resolveImage`, and consolidated with the other part there), and `pullImage` could expect input in the `ParseImageName` format.

(This is a pre-existing situation, so non-blocking; the changes to `ResolveName` in this PR now allow making this unambiguous.)

Reply:
Lets handle this in a separate PR